### ISSUE TITLE
[BUG] 'npm update' results in unclear error on peer-dep mismatch

### ISSUE THREAD

**Issue by KilianKilmister** [npm/cli#1681]
[Companion issue posted in arborist](https://github.com/npm/arborist/issues/114)

### Current Behavior:

```npm update``` results in arborist thowing an Error on mismatched peer-dep

the resulting error message doesn't point at what caused the error:
> npm ERR! code ERESOLVE
npm ERR! unable to resolve dependency tree
>
>npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kilianhefti/.npm/_logs/2020-08-16T01_03_09_094Z-debug.log
Waiting for the debugger to disconnect...

### Expected Behavior:

Concise error message ~*(and a way of strong-arming a match as no dev-/preview/-whatever-version will ever be compatible)*~ [RRFC about current peer-dep behaviour]

### Steps To Reproduce:
have unmatched peer-dep (eg. ts-node and typescript@4.0.0-dev)

### Environment:
final log entries: *dist, node-v and npm-v are highlited*
>1001 timing command:update Completed in 20505ms
1002 verbose type peer
1003 verbose stack Error: unable to resolve dependency tree
1003 verbose stack     at Arborist.[placeDep] (/Users/kilianhefti/.npm-packages/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:771:27)
1003 verbose stack     at /Users/kilianhefti/.npm-packages/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:588:46
1003 verbose stack     at Array.map (<anonymous>)
1003 verbose stack     at Arborist.[buildDepStep] (/Users/kilianhefti/.npm-packages/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:588:8)
1003 verbose stack     at async Arborist.buildIdealTree (/Users/kilianhefti/.npm-packages/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:160:7)
1003 verbose stack     at async Promise.all (index 1)
1003 verbose stack     at async Arborist.reify (/Users/kilianhefti/.npm-packages/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:126:5)
1003 verbose stack     at async update (/Users/kilianhefti/.npm-packages/lib/node_modules/npm/lib/update.js:34:3)
1004 verbose cwd /Users/kilianhefti/dev/cmd-repo
1005 verbose **Darwin 19.6.0**
1006 verbose argv "/usr/local/bin/node" "/Users/kilianhefti/.npm-packages/bin/npm" "update"
1007 verbose **node v14.8.0**
1008 verbose **npm  v7.0.0-beta.4**
1009 error code ERESOLVE
1010 error unable to resolve dependency tree
1011 verbose exit [ 1, true ]

### Note

~I will file a propper **RFC** about strong-arming the peer-dep stuff *(Link to come)*~
[RRFC about current peer-dep behaviour]
But having had to debug dependency resolution to discover the issue, I have to say, there is some real nice code in the **npm-repos**. well commented, clean, smart. **v7** is looking good.
And i've been experimenting with **npm-workspaces**: the linking works like a charm and everything looks really clean. there should be some beta documentation on them tho. i only descovered that it was working while i debugged **Arborists** tree-building.

[RRFC about current peer-dep behaviour]:<https://github.com/npm/rfcs/issues/204>

**Comment by ruyadorno** [comment]
hi @KilianKilmister thanks for taking the time to report 😊 as you have been following along the story around peer dependencies has been evolving a lot and quite quickly during the beta period, since you reported this the `ERESOLVE` has been expanded to contain a human-readable explanation for the conflict and with this week's pre-release (`v7.0.0-beta.12`) we flipped the switch and the default behavior for `npm install` is to be more resilient and only WARN in cases where the install tree can still be constructed (the previous more strict behavior is now behind a `--strict-peer-deps` flag).

We would love to see more feedback and bug reports if you get some time to give it another try to the beta. Thanks again for all the help!

**Comment by KilianKilmister** [comment]
@ruyadorno  Apologies on the late reply, I've been having way too much on my plate. While i did read it just after i got the notification, writing the reply got drowned out by other things.

I did update npm right after you posted the reply and it works as advertised.
I've been using the beta as my main npm-client (updating at each prerelease) since it was opened and i must say i'm very happy with both its initial state and the in-beta progress. v7 adresses many of the criticisms i had of npm as a tool and the new features have new allowed me to use mono-repos as the prefered structure for new projects. With this change, the last major problem i had with v7 has now been solved aswell.

I invested a lot of work on the topic of peer-deps over at the RFC repo in the weeks after i opened this issue and after a last minute removal from the task list of an openRFC meeting and not Having gotten a reply on the followup of that, I was sure that my efforts have been in vain.
I am really, really glad that parts of it seem to have stuck around non-the-less and i was very relieved when i saw this change.

Thanks a lot.
